### PR TITLE
[sc-28554] implement removing files from a los bundle

### PR DESCRIPTION
**Pull request by Greek**
Updates the implementation for toggling file bundle states on the client, see https://app.shortcut.com/clarkcan/story/28554/implement-removing-files-from-a-los-bundle-into-clark-service 

Example:
https://github.com/Cyber4All/clark-client/assets/17275723/2dd0a253-1984-4347-aefc-dac32aaf8015

### CODE REVIEW

**Comment by Greek** [comment]
i don't think i can test this properly till we get learning-object module working again. for now i'll mark the story as blocked and try it again later

File: src/entity/learning-object/learning-object.ts [hunk 1]
```diff
@@ -838,7 +838,8 @@ export namespace LearningObject {
 
   export namespace Material {
     export interface File {
-      id: string;
+      _id: string;
+      id: string; // Keeping for backward compat, may wanna remove?
```
Comment (shanioffe) [inline src/entity/learning-object/learning-object.ts:842]: Hmm, unless someone says definitely to remove it, you could mark `id` as deprecated, or you could change its name to something like "source" like the way [learning-outcome model](https://github.com/Cyber4All/clark-service/blob/main/src/shared/types/learning-outcome.type.ts) is implemented.

File: src/app/onion/core/learning-object.service.ts [hunk 2]
```diff
@@ -51,26 +51,24 @@ export class LearningObjectService {
...
-    const route = BUNDLING_ROUTES.TOGGLE_BUNDLE_FILE({ username, learningObjectId });
+    const route = BUNDLING_ROUTES.TOGGLE_BUNDLE_FILE({ learningObjectId });
 
     return this.http
       .patch(
         route,
         {
           fileIDs: fileIDs,
-          state: state
+          packagable: state
```
Comment (mfranz1) [inline src/app/onion/core/learning-object.service.ts:71]: thank you

File: src/entity/learning-object/learning-object.ts (hunk 1 above)
Comment (mfranz1) [reply src/entity/learning-object/learning-object.ts:842]: I believe it is safe to remove `id`. Please just double check where the File interface is being implemented/referenced throughout the client.

**Review by curlyLasagna — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

File: src/entity/learning-object/learning-object.ts (hunk 1 above)
Comment (Greek) [reply src/entity/learning-object/learning-object.ts]: done. the client builds, i believe this change should be okay to deploy =)

**Review by mfranz1 — APPROVED** [review]

**Review by shanioffe — APPROVED** [review]

**Pull request merged**